2023-11-05 20:54:16,690 INFO - __main__ - Finished setting up the logger.
2023-11-05 20:54:16,918 INFO - numexpr.utils - NumExpr defaulting to 8 threads.
2023-11-05 20:54:17,168 INFO - __main__ - Finished loading the dataset.
2023-11-05 20:54:17,171 CRITICAL - utils.logger - Unhandled exception.
Traceback (most recent call last):
  File "/Users/valen/Documents/GitHub/golem/golem_PT/src/main.py", line 70, in <module>
    main()
  File "/Users/valen/Documents/GitHub/golem/golem_PT/src/main.py", line 56, in main
    B_est = golem(dataset.X, args.lambda_1, args.lambda_2, args.equal_variances, args.num_iter,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/valen/Documents/GitHub/golem/golem_PT/src/golem.py", line 44, in golem
    model = GolemModel(n, d, lambda_1, lambda_2, equal_variances, seed, B_init)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/valen/Documents/GitHub/golem/golem_PT/src/models/golem_model.py", line 39, in __init__
    self._build()
  File "/Users/valen/Documents/GitHub/golem/golem_PT/src/models/golem_model.py", line 47, in _build
    self.B = nn.Parameter(torch.zeros([self.d, self.d], dtype=torch.float32))
    ^^^^^^
  File "/Users/valen/anaconda3/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1709, in __setattr__
    raise AttributeError(
AttributeError: cannot assign parameters before Module.__init__() call
